File: lv2/Python3/올바른괄호.py
def solution(s):
    if s[0] == ')' or s[-1] == '(': return False
    stack = [s[0]]

    for i in range(1, len(s)):
        if s[i] == '(':
            stack.append(s[i])
        else:
            if stack:
                stack.pop()
            else:
                return False
    if len(stack) == 0: return True
    else: return False

File: lv2/Java/올바른괄호.java
package lv2.Java;

import java.util.Stack;

public class 올바른괄호 {
    boolean solution(String s) {
        Stack<Integer> stack = new Stack<>();
        if(s.charAt(0) == ')' || s.charAt(s.length()-1) == '(') return false;

        stack.push(1);
        for (int i = 1; i < s.length(); i++) {
            if(s.charAt(i) == '(') stack.push(1);
            else{
                if(stack.size() == 0) return false;
                if(stack.pop() != 1) return false;
            }
        }
        if(stack.size() == 0) return true;
        else return false;

    }
}
